File: Part3/T3L100extra_ok.c
long int I,J;
float F;
unsigned char C;
long int Iaa[14][12];  // starting indices: 0, 0
double Raaa[3][6][4]; // starting indices 2, 0, 1
double R;


void Print_nl();
void Print_dot();
void Print_even();
void Print_odd();
void Print_j();
void Print_i_j_values();
void Print_oops();

int main()
{
    for (I = 1; I <= 10; I++)
        Print_dot();
    Print_nl();

    for (I = 1; I <= 100; I++) {
        Print_dot();
        if (I == 10) break;
    }
    
    Print_nl();

    I = 15;
    for (I = 1; I <= 15; I++)
        Print_dot();
    Print_nl();

    for (C = 1; C <= 10; C++)
        Print_dot();
    Print_nl();

    for (I = 0; I <= 9; I++) {
        switch (I) {
            case 0:
            case 2:
            case 4:
            case 6:
            case 8:
                Print_even(); break;
            case 1:
            case 3:
            case 5:
            case 7:
            case 9:
                Print_odd(); break;
        }
        if (I == 1) I = 3;
        for (J = 1; J <= I-1; J++)
            if (J == 7) break;
            else Print_j();
        if (I > 99) break;
    }
    Print_nl();
    Print_i_j_values();

    for (J = 10; J >= 1; J--)
        Print_i_j_values();

    for (I = 0; I <= 9; I++);
    Print_oops();

    for (I = 10; I >= 0; I--) {
        switch (I) {
            case 0:
            case 2:
            case 4:
            case 6:
            case 8:
            case 10:
                Print_even(); break;
            case 1:
            case 3:
            case 7:
            case 9:
                Print_odd(); break;
            case 5:
                break;
        }
        if (I == 5) break;
        Print_oops();
    }

    return 0;
}

File: test/T3L100extra_ok.pas
(* Test file for 100% level of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L100_ok;

var
   i,j  : Integer;
   f    : Single;
   c    : Char;
   iaa  : array[0..13, 0..11] of Integer;
   raaa : array[2..4, 0..5, 1..4] of Real;
   r    : Real;


procedure print_nl; External;
procedure print_dot; External;
procedure print_even; External;
procedure print_odd; External;
procedure print_j; External;
procedure print_i_j_values; External;
procedure print_oops; External;


begin
   for i := 1 to 10 do
      print_dot;
   print_nl;

   for i := 1 to 100 do begin
      print_dot;
      if i = 10 then break
   end;
   print_nl;

   i := 15;
   for i := 1 to i do
      print_dot;
   print_nl;

   for c := chr(1) to chr(10) do
      print_dot;
   print_nl;

   for i := 0 to 9 do begin
      case i of
	0,2,4,6,8: print_even;
	1,3,5,7,9: print_odd
      end;
      if i = 1 then i := 3;
      for j := 1 to i-1 do
	 if j = 7 then break
	 else print_j;
      if i > 99 then break
   end;
   print_nl;
   print_i_j_values;

   for j := 10 downto 1 do
      print_i_j_values;

   for i := 0 to 9 do;
   print_oops;

   for i := 10 downto 0 do begin
      case i of
	0,2,4,6,8,10: print_even;
	1,3,7,9     : print_odd;
	5           : break;
      end; { case }
      print_oops
   end

end.
